apps/src/javalab/JavabuilderConnection.js

```diff
@@ -104,6 +104,19 @@ export default class JavabuilderConnection {
 
   // Send a message across the websocket connection to Javabuilder
   sendMessage(message) {
-    this.socket.send(message);
+    if (!this.socket) {
+      console.error(`[error] The connection has closed.`);
```

Comment:
should we surface an error to the user here? could this happen if the user types into the console but their program isn't running?

Reply:
Wouldn't there be an error anyway because this function relies on there being a socket already?